main.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+"""
+Example of using Rerun to log and visualize the out of grounded dino + segment-anything.
+
+See: [segment_anything](https://github.com/IDEA-Research/Grounded-Segment-Anything).
+
+Can be used to test mask-generation on one or more images, as well as videos. Images can be local file-paths
+or remote urls. Videos must be local file-paths. Can use multiple prompts.
+"""
+
+
+import argparse
+import logging
+import rerun as rr
+import torch
+import cv2
+from pathlib import Path
+from models import CONFIG_PATH, MODEL_URLS, get_downloaded_model_path
+from models import load_grounding_model, create_sam, load_image, image_to_tensor
+from models import get_grounding_output, run_segmentation, resize_img
+from segment_anything import SamPredictor
+from segment_anything.modeling import Sam
+from groundingdino.models import GroundingDINO
+
+
+def log_images_segmentation(args, model: GroundingDINO, predictor: Sam):
+    for n, image_uri in enumerate(args.images):
+        rr.set_time_sequence("image", n)
+        for prompt in args.prompts:
+            image, image_tensor = load_image(image_uri)
+            # run grounding dino model
+            logging.info(f"Running GroundedDINO with DETECTION PROMPT {prompt}.")
+            boxes_filt, pred_phrases = get_grounding_output(
+                model, image_tensor, prompt, 0.3, 0.25, device=args.device
+            )
+            # denormalize boxes (from [0, 1] to image size)
+            H, W, _ = image.shape
+            for i in range(boxes_filt.size(0)):
+                boxes_filt[i] = boxes_filt[i] * torch.Tensor([W, H, W, H])
+                boxes_filt[i][:2] -= boxes_filt[i][2:] / 2
+                boxes_filt[i][2:] += boxes_filt[i][:2]
+
+            predictor.set_image(image)
+            run_segmentation(predictor, image, boxes_filt, prompt)
+
+
+def log_video_segmentation(args, model: GroundingDINO, predictor: Sam):
+    video_path = args.video_path
+    assert video_path.exists()
+    cap = cv2.VideoCapture(str(video_path))
+
+    prompt = args.prompts
+
+    idx = 0
+    while cap.isOpened():
+        ret, bgr = cap.read()
+        if not ret:
+            break
+        rr.set_time_sequence("frame", idx)
+        rgb = cv2.cvtColor(bgr, cv2.COLOR_BGR2RGB)
+        rgb = resize_img(rgb, 512)
+
```

Comment:
```suggestion
            predictor.set_image(rgb)
```
Set the image once per image